kernel_tuner/nvml.py

```diff
@@ -137,12 +138,12 @@ def gr_clock(self, new_clock):
 
     @property
     def mem_clock(self):
-        """Control the graphics clock (may require permission), only values compatible with the graphics clock can be set directly"""
+        """Control the memory clock (may require permission), only values compatible with the graphics clock can be set directly"""
         return pynvml.nvmlDeviceGetClockInfo(self.dev, pynvml.NVML_CLOCK_MEM)
 
     @mem_clock.setter
     def mem_clock(self, new_clock):
-        if new_clock != pynvml.nvmlDeviceGetApplicationsClock(self.dev, pynvml.NVML_CLOCK_MEM):
+        if new_clock != pynvml.nvmlDeviceGetMaxClockInfo(self.dev, pynvml.NVML_CLOCK_MEM):
```

Comment:
The if statement here is mostly there to check if the new_clock is not already the currently used "applications clock" setting on the device, the main goal is to avoid having to call nvidia-smi for an operation that is not necessary. Are you sure that nvmlDeviceGetMaxClockInfo gives the "applications clock"? If I read the documentation correctly it seems that nvmlDeviceGetMaxClockInfo reports the actual maximum clock of the GPU.

Reply:
I reverted the code back to use `nvmlDeviceGetApplicationsClock`. That indeed seems like the correct function to call in this context.